deepwell/src/endpoints/parent.rs

```diff
@@ -85,3 +88,90 @@ pub async fn parent_remove(
 
     ParentService::remove(ctx, input).await
 }
+
+pub async fn parent_get_all(
+    ctx: &ServiceContext<'_>,
+    params: Params<'static>,
+) -> Result<Vec<String>> {
+    let GetPageReference { site_id, page } = params.parse()?;
+
+    info!(
+        "Getting parents for child {:?} in site ID {}",
+        page, site_id,
+    );
+
+    let parents: Vec<Reference<'_>> = ParentService::get_parents(ctx, site_id, page)
+        .await?
+        .iter()
+        .map(|p| Reference::from(p.parent_page_id))
+        .collect();
+
+    let pages: Vec<String> = PageService::get_pages(ctx, site_id, parents.as_slice())
+        .await?
+        .iter()
+        .map(|p| p.slug.clone())
+        .collect();
+
+    Ok(pages)
+}
+
+pub async fn parent_modify(
```

Comment:
I think `update` is a better name, like get, create, update, delete as the basic ops.